crates/turbopack-ecmascript/js/src/runtime.js

```diff
@@ -628,8 +630,17 @@ function disposePhase(outdatedModules, disposedModules) {
     disposeModule(moduleId, "clear");
   }
 
+  const outdatedModuleParents = new Map();
+  for (const moduleId of outdatedModules) {
+    const oldModule = moduleCache[moduleId];
+    outdatedModuleParents.set(moduleId, oldModule?.parents);
+    delete moduleCache[moduleId];
```

Comment:
Can you add a comment explaining that we need the `disposeModule` calls above to run so that parents are updated accordingly, before we can take them out of the cache?

I'm wary of having this logic both here and in `disposeModule`. Is there a way we could keep this in a single function somewhere? This seems very fragile.